crates/cext/src/transpiler/transpile_function.rs

```diff
@@ -0,0 +1,200 @@
+    /// The seed for the transpiler. If set to a negative number this means no seed will be
+    /// set and the RNGs used in the transpiler will be seeded from system entropy.
+    seed: i64,
+    /// The approximation degree a heurstic dial where 1.0 means no approximation (up to numerical
+    /// tolerance) and 0.0 means the maximum approximation. A `NAN` value indicates that
+    /// approximation is allowed up to the reported error rate for an operation in the target.
+    approximation_degree: f64,
+}
+
+impl Default for TranspileOptions {
+    fn default() -> Self {
+        TranspileOptions {
+            optimization_level: 2,
+            seed: -1,
+            approximation_degree: 1.0,
+        }
+    }
+}
+
+/// @ingroup QkTranspiler
+///
+/// Generate transpiler options defaults
+///
+/// This function generates a QkTranspileOptions with the default settings
+/// This currently is ``optimization_level`` 2, no seed, and no approximation.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub extern "C" fn qk_transpiler_default_options() -> TranspileOptions {
+    TranspileOptions::default()
+}
+
+/// @ingroup QkTranspiler
+/// Transpile a single circuit
+///
+/// The Qiskit transpiler is a quantum circuit compiler that rewrites a given
+/// input circuit to match the constraints of a QPU and/or optimize the circuit
+/// for execution. This function should only be used with circuit's constructed solely
+/// using Qiskit's C API, it makes assumptions on the circuit only using features exposed via C,
+/// if you are in a mixed Python and C environment it is typically better to invoke the transpiler
+/// via Python.
+///
+/// This function is multithreaded internally and will launch a thread pool
+/// with threads equal to the number of CPUs reported by the operating system by default.
+/// This will include logical cores on CPUs with SMT. You can tune the number of threads
+/// with the ``RAYON_NUM_THREADS`` environment variable. For example, setting
+/// ``RAYON_NUM_THREADS=4`` would limit the thread pool to 4 threads.
+///
+/// @param circuit A pointer to the circuit to run the transpiler on
+/// @param target A pointer to the target to compile the circuit for
+/// @params options A pointer to an options object that defines user options. If this is a null
+///   pointer the default values will be used. See ``qk_transpile_default_options``
+///   for more details on the default values.
+/// @param result A pointer to the memory location of the transpiler result. On a successful
+///   execution (return code 0) the output of the transpiler will be written to the pointer. The
+///   members of the result struct are owned by the caller and you are responsible for freeing
+///   the members using the respective free functions.
+/// @param error A pointer to a pointer with an nul terminated string with an error description.
+///   If the transpiler fails a pointer to the string with the error description will be written
```

Comment:
```suggestion
///   If the transpiler fails, a pointer to the string with the error description will be written
```
Also, should we adopt this pattern with the standalone passes in C?

Reply:
I think after we resolve the error handling story for C. Right now this pattern is kind of future facing and we just stuff a backtrace in the error string as that's the best we can do.